srml/contracts/src/exec.rs

```diff
@@ -398,9 +411,13 @@ where
 		let (change_set, events, calls) = {
 			let mut overlay = OverlayAccountDb::new(&self.overlay);
 
-			overlay.create_contract(&dest, code_hash.clone())?;
+			let contract = overlay.create_contract(&dest, code_hash.clone())?;
 
-			let mut nested = self.nested(overlay, dest.clone());
+			let mut nested = self.nested(
+				overlay,
+				dest.clone(),
+				Some(ContractInfo::Alive(contract))
+			);
```

Comment:
as in the other comment we now this new contract doesn't have an actual contract info in storage thus we can actually call nested with None